CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -297,16 +297,18 @@ public static function updateCacheTime($groupID, $processed) {
    * In fact it turned out there is little overlap between the code when group is passed in
    * and group is not so it makes more sense as separate functions.
    *
-   * @todo remove last call to this function from outside the class then make function protected,
-   * enforce groupID as an array & remove non group handling.
+   * @todo enforce groupID as an array & remove non group handling.
+   * Use flushCaches when no group id provided.
    *
    * @param int $groupIDs
    *   the groupID to delete cache entries, NULL for all groups.
    * @param bool $onceOnly
    *   run the function exactly once for all groups.
    */
-  public static function remove($groupIDs = NULL, $onceOnly = TRUE) {
-    static $invoked = FALSE;
+  protected static function remove($groupIDs = NULL, $onceOnly = TRUE) {
```

Comment:
A quick `grep` against `universe` turns up a contrib caller who uses this:

```
$ svngrep GroupContactCache::remove universe
...*snip*...
universe/civihr/hrjobcontract/CRM/Contact/Form/Inline/CustomData.php:    CRM_Contact_BAO_GroupContactCache::remove();
```

@eileenmcnaughton @MegaphoneJon - could we keep the contract? Or perhaps you have a suggestion for @davialexandre etal on updating that?

Reply:
Using the api job.group_cache_flush is supported & unit tested.

Reply:
I have updated it to be non-fatal for any extensions calling this from outside of core (note that we don't guarantee the stability of core functions other than the api )